CMakeLists.txt

```diff
@@ -29,83 +37,129 @@ message( STATUS "-------- cpp2py version and git hash detection -------------")
 set(LIBCLANG_LOCATION "${LIBCLANG_LOCATION}" CACHE STRING "Location of the libclang library")
 
-# --- Python ---
+#--------------------------------------------------------
+#   Python and Numpy
+#--------------------------------------------------------
+find_package(Python REQUIRED COMPONENTS Interpreter Development NumPy)
+
+# check required python packages
+function(exec_python_script the_script output_var_name)
+  execute_process(COMMAND ${Python_EXECUTABLE} -c "${the_script}"
+    OUTPUT_VARIABLE res RESULT_VARIABLE returncode OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(NOT returncode EQUAL 0)
+    message(FATAL_ERROR "The script : ${the_script} \n did not run properly in the Python interpreter. Check your python installation.")
+  endif()
+  set(${output_var_name} ${res} PARENT_SCOPE)
+endfunction()
+
+exec_python_script("import mako.template" nulle)
+exec_python_script("import numpy" nulle)
+exec_python_script("import scipy" nulle)
+if(Build_Documentation)
+  exec_python_script("import clang.cindex" nulle)
+endif()
+
+# set variables used in the current and in dependent projects
+set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE STRING "Python executable")
+if(Python_SOABI)
+  set(PYTHON_MODULE_EXT .${Python_SOABI}.so CACHE STRING "Python module extension")
+else()
+  set(PYTHON_MODULE_EXT .so CACHE STRING "Python module extension")
+endif()
+if(BUILD_DEBIAN_PACKAGE)
+  set(PYTHON_LIB_DEST_DIR dist-packages)
+else()
+  set(PYTHON_LIB_DEST_DIR site-packages)
+endif()
+set(PYTHON_LIB_DEST_ROOT lib/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/${PYTHON_LIB_DEST_DIR} CACHE PATH "Python module directory")
 
-find_package(Python)
-set(CPP2PY_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE} CACHE FILEPATH "Python Executable")
-set(Python_EXECUTABLE ${PYTHON_EXECUTABLE} CACHE FILEPATH "Python Executable")
+set(CPP2PY_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE} CACHE FILEPATH "Python executable")
 set(CPP2PY_PYTHON_LIB_DEST_ROOT ${PYTHON_LIB_DEST_ROOT} CACHE FILEPATH "Python module directory")
 set(CPP2PY_PYTHON_MODULE_EXT ${PYTHON_MODULE_EXT} CACHE FILEPATH "Extension of compiled Python modules")
+
+# output the python variables
+message(STATUS "Python_EXECUTABLE (= PYTHON_EXECUTABLE): " ${Python_EXECUTABLE})
+message(STATUS "Python_VERSION: " ${Python_VERSION})
+message(STATUS "Python_VERSION_MAJOR: " ${Python_VERSION_MAJOR})
+message(STATUS "Python_VERSION_MINOR: " ${Python_VERSION_MINOR})
+message(STATUS "Python_INCLUDE_DIRS: " ${Python_INCLUDE_DIRS})
+message(STATUS "Python_NumPy_INCLUDE_DIRS: " ${Python_NumPy_INCLUDE_DIRS})
+message(STATUS "Python_NumPy_VERSION: " ${Python_NumPy_VERSION})
+message(STATUS "Python_SITEARCH: " ${Python_SITEARCH})
+message(STATUS "Python_STDLIB: " ${Python_STDLIB})
+message(STATUS "Python_LIBRARIES: " ${Python_LIBRARIES})
+message(STATUS "Python_SOABI: " ${Python_SOABI})
+message(STATUS "PYTHON_MODULE_EXT: " ${PYTHON_MODULE_EXT})
+message(STATUS "Python modules install path: ${CMAKE_INSTALL_PREFIX}/${PYTHON_LIB_DEST_ROOT}")
```

Comment:
We don't need all these prints?

Reply:
I think it can be helpful to see what is found by the find_package call, but we can remove it if you prefer.